include/libtorrent/kademlia/node_entry.hpp

```diff
@@ -69,8 +69,7 @@ struct TORRENT_EXTRA_EXPORT node_entry
 
 	node_id id;
 
-	address_v4::bytes_type a;
-	boost::uint16_t p;
+	union_endpoint endpoint;
```

Comment:
I'm a little bit worried about this blowing up memory usage for the IPv4 routing table. Do you think it would make sense to make this type a template, with the address::bytes_type as a template parameter?